aries_cloudagent/resolver/base.py

```diff
@@ -61,19 +61,17 @@ def supports(self, method: str) -> bool:
 
     async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
         """Resolve a DID using this resolver."""
-        if isinstance(did, str):
-            did = DID(did)
+        py_did = DID(did) if isinstance(did, str) else did
 
-        if not self.supports(did.method):
+        if not self.supports(py_did.method):
             raise DIDMethodNotSupported(
-                f"{self.__class__.__name__} does not support DID method {did.method}"
+                f"{self.__class__.__name__} does not support DID method {py_did.method}"
             )
 
-        did = str(did)
-        did_document = await self._resolve(profile, did)
+        did_document = await self._resolve(profile, str(py_did))
         result = DIDDocument.deserialize(did_document)
         return result
 
     @abstractmethod
-    async def _resolve(self, profile: Profile, did: DID) -> dict:
+    async def _resolve(self, profile: Profile, py_did: DID) -> dict:
```

Comment:
This line is actually a mistake; the type for `did` should be `str` with the intention being that resolver plugins need not concern themselves with PyDID if they don't want to. Sorry for the mix-up.

Reply:
It's no trouble, let's just converge on what's right